src/ArrayOfArrays.hpp

```diff
@@ -108,10 +108,16 @@ class ArrayOfArrays : protected ArrayOfArraysView< T, INDEX_TYPE >
 
   /**
    * @brief Default move assignment operator, performs a shallow copy.
+   * @param src The ArrayOfArrays to be moved from.
    * @return *this
    */
   inline
-  ArrayOfArrays & operator=( ArrayOfArrays && ) = default;
+  ArrayOfArrays & operator=( ArrayOfArrays && src )
+  {
```

Comment:
This was a memory leak, since the original data contained in `*this` wasn't free'd.